### PR TITLE
[Perf] Change Trtllm fp8 MoE to use Shuffled Weights and BlockMajorK Layout

### PR DESCRIPTION
**Pull request by wzhao18**
## Purpose
Change Trtllm fp8 MoE to use Shuffled weights and BlockMajorK layout. Adapted from examples in https://github.com/flashinfer-ai/flashinfer/blob/19329d838236803c59c8f571d42910b1c0c2a18e/tests/moe/test_trtllm_gen_fused_moe.py#L838

Benchmarking shows this improves performance across the board.

## Test Plan
Ensure model evals look good for both the modular (MiniMaxAI/MiniMax-M2.5) and monolithic (deepseek-ai/DeepSeek-R1) kernel code paths.

## Test Result
```
vllm serve MiniMaxAI/MiniMax-M2.5  --trust-remote-code --tensor-parallel-size 2

|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.9356|±  |0.0068|
|     |       |strict-match    |     5|exact_match|↑  |0.9340|±  |0.0068|
```

With #38989
```
vllm serve deepseek-ai/DeepSeek-R1 --trust-remote-code --tensor-parallel-size 8

|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value|   |Stderr|
|-----|------:|----------------|-----:|-----------|---|----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.953|±  |0.0058|
|     |       |strict-match    |     5|exact_match|↑  |0.950|±  |0.0060|
```

## Benchmark Result
Kernel microbenchmark:
```
     M  variant                  kernel(ms)                                                                                                                                           
  ------  ----------------------  -----------                                                                                                                                         
       1  no_shuffle                    0.028                                                                                                                                         
       1  shuffle_majork                0.028                                                                                                                                         
       1  shuffle_blockmajork           0.028                                                                                                                                         
  ------  ----------------------  -----------                                                                                                                                         
       8  no_shuffle                    0.086                                                                                                                                         
       8  shuffle_majork                0.083                                                                                                                                         
       8  shuffle_blockmajork           0.081                                                                                                                                         
  ------  ----------------------  -----------                                                                                                                                       
      64  no_shuffle                    0.249                                                                                                                                         
      64  shuffle_majork                0.249                                                                                                                                         
      64  shuffle_blockmajork           0.247                                                                                                                                         
  ------  ----------------------  -----------                                                                                                                                         
     128  no_shuffle                    0.288                                                                                                                                         
     128  shuffle_majork                0.288                                                                                                                                         
     128  shuffle_blockmajork           0.281                                                                                                                                         
  ------  ----------------------  -----------                                                                                                                                         
     256  no_shuffle                    0.313                                                                                                                                         
     256  shuffle_majork                0.312                                                                                                                                         
     256  shuffle_blockmajork           0.299                                                                                                                                       
  ------  ----------------------  -----------                                                                                                                                         
     512  no_shuffle                    0.474                                                                                                                                         
     512  shuffle_majork                0.464                                                                                                                                         
     512  shuffle_blockmajork           0.467                                                                                                                                         
  ------  ----------------------  -----------                                                                                                                                         
    1024  no_shuffle                    0.530                                                                                                                                       
    1024  shuffle_majork                0.511                                                                                                                                         
    1024  shuffle_blockmajork           0.513                                                                                                                                         
  ------  ----------------------  -----------                                                                                                                                         
    2048  no_shuffle                    0.640                                                                                                                                         
    2048  shuffle_majork                0.615                                                                                                                                         
    2048  shuffle_blockmajork           0.589                                                                                                                                         
  ------  ----------------------  -----------
```

E2E benchmark:
Minimax M2.5 TP=2
isl1024_osl1024
isl8192_osl1024

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

**BEFORE SUBMITTING, PLEASE READ <https://docs.vllm.ai/en/latest/contributing>** (anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

File: vllm/model_executor/warmup/deep_gemm_warmup.py [hunk 1]
```diff
@@ -168,13 +168,13 @@ def _fused_moe_grouped_gemm_may_use_deep_gemm(module: torch.nn.Module) -> bool:
     ):
         return False
 
-    if not isinstance(module.quant_method, FusedMoEModularMethod):
-        # modular kernels could invoke deep_gemm_moe_fp8
-        return True
+    moe_kernel = getattr(module.quant_method, "moe_kernel", None)
```
Comment (robertgshaw2-redhat) [inline vllm/model_executor/warmup/deep_gemm_warmup.py:171]: just wondering why this is touched?

Comment (wzhao18) [reply vllm/model_executor/warmup/deep_gemm_warmup.py:171]: `_fused_moe_grouped_gemm_may_use_deep_gemm` returns True despite trtllm MoE is used. In this warmup call, it has an assertion on the MoE weight shape, which fails as the weight layout is changed in this PR.

Comment (robertgshaw2-redhat) [reply vllm/model_executor/warmup/deep_gemm_warmup.py:171]: nice, I wonder if we have been doing additonal warmup accidentally 🫣

**Review by robertgshaw2-redhat — APPROVED** [review]
